contrib/auto-render/auto-render.js

```diff
@@ -83,30 +85,26 @@ const defaultOptions = {
     ],
 };
 
-const extend = function(obj) {
-    // Adapted from underscore.js' `_.extend`. See LICENSE.txt for license.
-    let source;
-    let prop;
-    const length = arguments.length;
-    for (let i = 1; i < length; i++) {
-        source = arguments[i];
-        for (prop in source) {
-            if (Object.prototype.hasOwnProperty.call(source, prop)) {
-                obj[prop] = source[prop];
-            }
-        }
-    }
-    return obj;
-};
+function AutoRenderSettings(options) {
+    // allow null options
+    options = options || {};
+    this.delimiters = options.delimiters || defaultOptions.delimiters;
+    this.ignoredTags = options.ignoredTags || defaultOptions.ignoredTags;
+}
 
 const renderMathInElement = function(elem, options) {
     if (!elem) {
         throw new Error("No element provided to render");
     }
 
-    options = extend({}, defaultOptions, options);
+    const autoRenderOptions = new AutoRenderSettings(options);
```

Comment:
This line will add keys to the passed-in `options` object.  It needs to go *after* the next line that makes the copy.